config/config.go

```diff
@@ -149,12 +141,21 @@ func loadConfig() (*config, error) {
 		configFileError = err
 	}
 
+	// Warn about missing Config file only after all other configuration is
+	// done.  This prevents the warning on help messages and invalid
+	// options.  Note this should go directly before the return.
+	if configFileError != nil {
+		log.Warning("%v", configFileError)
+	}
+
+	// TODO: move out of here
 	// Finally, parse the remaining command line options again to ensure
 	// they take precedence.
-	if _, err := flags.Parse(&cfg); err != nil {
-		return nil, err
-	}
+	return ParseFlags(cfg)
```

Comment:
This imho should just return the config and not call `ParseFlags` (since it is already called in `poet.go` before this function is called.

The comment above suggests that this is intentional, but I don't think it is?

Reply:
It is intentional. It makes sure that command-line flags are "on top". 

Moved additional ParseFlags out to `poetMain()`.